lib/view/templates.py

```diff
@@ -1855,38 +1814,135 @@ def turn_empty_to_none(ls):
-                        extract_value_from_dict("single-record-tps"),
-                        Projectors.Identity("data", "write-info"),
-                    ),
+            ),
+        ),
+    ),
+    from_source="data",
+    for_each="data",
+    order_by=FieldSorter("User"),
+)
+
+
+def create_quota_weighted_avg(type: str):
+    """
+    A simple averaging would work fine since all nodes likely have the same quota. Although,
+    there is a slight chance they do not.
+    """
+
+    def usage_weighted_avg(edatas: list[EntryData]):
+        pcts: map[float] = map(lambda edata: edata.value, edatas)
+        weights: list[float] = list(
+            map(lambda edata: edata.record[type]["Quota"], edatas)
+        )
+
+        if not weights:
+            return None
+
+        return weighted_avg(pcts, weights)
+
+    return usage_weighted_avg
+
+
+def create_quota_tps_subgroup(
+    type: Literal["Read"] | Literal["Write"],
+    key: Literal["read-info"] | Literal["write-info"],
+):
+    return Subgroup(
+        type,
+        (
+            Field(
+                "Quota",
+                Projectors.Func(
+                    FieldType.number,
+                    extract_value_from_dict("quota"),
+                    Projectors.Identity("data", key),
                 ),
-                # TODO: Support for Subgroups to have Subgroups to have Scan/Query be a
-                # subgroup of Write
-                Field(
-                    "Scan/Query Limited RPS",
-                    Projectors.Func(
-                        FieldType.undefined,
-                        extract_value_from_dict("scan-query-rps-limited"),
-                        Projectors.Identity("data", "write-info"),
+                converter=Converters.scientific_units,
+                aggregator=Aggregators.sum(converter=Converters.scientific_units),
+            ),
+            Field(
+                "Usage%",
+                Projectors.Div(
+                    Projectors.Sum(
+                        Projectors.Func(
+                            FieldType.number,
+                            extract_value_from_dict("single-record-tps"),
+                            Projectors.Identity("data", key),
+                        ),
+                        Projectors.Func(
+                            FieldType.number,
+                            extract_value_from_dict("scan-query-rps-limited"),
+                            Projectors.Identity("data", key),
+                        ),
                     ),
-                ),
-                Field(
-                    "Scan/Query Limitless",
                     Projectors.Func(
-                        FieldType.undefined,
-                        extract_value_from_dict("scan-query-limitless"),
-                        Projectors.Identity("data", "write-info"),
+                        FieldType.number,
+                        lambda x: _ignore_zero(extract_value_from_dict("quota")(x)),
```

Comment:
I don't understand this, does `extract_value_from_dict("quota")` return a callable that accepts `x`?

Reply:
```
def extract_value_from_dict(key: str):
    def extract_value(dict):
        if key not in dict:
            raise NoEntryException(f"{key} not found dict")

        return dict[key]

    return extract_value
 ```
 yes, and if that value is zero we want to ignore it

Reply:
x is a dict whose corresponding value is either "read-info" or "write-info"

Reply:
makes sense, thanks